src/snowflake/snowpark/_internal/analyzer/expression.py

```diff
@@ -292,6 +292,38 @@ def individual_node_complexity(self) -> Dict[PlanNodeCategory, int]:
         )
 
 
+class UnresolvedColumnRegex(Expression):
+    def __init__(self, expressions: List[Attribute]) -> None:
+        super().__init__()
+        self.expressions = expressions
+
```

Comment:
we could also add an assert here
```suggestion
        assert len(self.expressions) > 0
        self.expressions = expressions

```